### DISCUSSION TITLE
Blinking between tiles in SFML Demo

### DISCUSSION THREAD

**Discussion by DmitruMoth** [fallahn/tmxlite#146] [Q&A]
So, problem is here:
[map.zip](https://github.com/fallahn/tmxlite/files/5188838/maps.zip)
![blinking](https://user-images.githubusercontent.com/52855719/92487829-70816b00-f1f6-11ea-8f77-f3e37a8c40c6.gif)

**Comment by fallahn** [comment]
This is a well documented issue with SFML vertex arrays, rather than the map loader itself. This post has some details why it happens: https://en.sfml-dev.org/forums/index.php?topic=15338.0 (or just search for SFML tile tearing)

A good solution I've found is to render the map with a shader, like this https://github.com/SFML/SFML/wiki/Source%3A-ShaderTileMap

**Comment by DmitruMoth** [comment]
How can i implement this to existing sfml demo?

**Comment by fallahn** [comment]
Short answer: not easily, it would pretty much be a re-write. The example is really just that: an example. Have you tried some of the suggested solutions in the forum topic, such as rounding your view's position to the neareset whole pixel? You could also render the visible set of chunks to a render texture first, then use that to make a sprite which you can scroll.

**Comment by DmitruMoth** [comment]
Thanks, i will try to fix this problem
p.s. rounding make it worse

**Comment by ldornbusch** [comment]
I could solve this problem by avoiding having transparent pixels *around* solid tiles. 
Check this detailed picture:
![image](https://github.com/fallahn/tmxlite/assets/13638691/21cf737f-3206-4463-9222-09ab7d0c4753)
It is clearly visible that not all tiles show the flickering but only some tiles comparing this with your tilesheet you can see that the tile in question has transparent pixels to the left:
![image](https://github.com/fallahn/tmxlite/assets/13638691/c69c3b4b-e17e-479a-97fc-57aa8e9c3339)
I do not understand exactly how this happened, but I think that sometimes a rounding error or a blending happens between the transparent and the filled pixel. However, by simply using a non-transparent 1pixel border (or placing other tiles next to there), you can fix this.
